### PR TITLE
fix: Blank states sidebar

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR fixes all the blank states in the left pane of the editor.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/29696

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Streamlined the user interface by simplifying components and removing unnecessary permission checks.
  - Consolidated empty state messaging across different sections for a more unified user experience.

- **Style**
  - Updated iconography in the IDE sidebar for better visual representation.

- **Documentation**
  - Improved in-app messaging and descriptions to enhance user guidance.

- **Bug Fixes**
  - Fixed conditional rendering issues to ensure consistent display of empty states when no items are present.

### CODE REVIEW

**Review by hetunandu — DISMISSED** [review]

File: app/client/src/pages/Editor/IDE/PagesPane/EmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { Button, Flex, Text, Icon } from "design-system";
+
+interface EmptyStateProps {
+  icon: string;
+  description: string;
+  buttonText: string;
+  onClick?: () => void;
+}
+
+const EmptyState = ({
```
Comment (KelvinOm) [inline app/client/src/pages/Editor/IDE/PagesPane/EmptyState.tsx:11]: Shouldn't this component be part of ADS? It seems that all empty states should be consistent within the system. I think this component should be abstract and reusable.

Here is an example of a [similar component](https://react-spectrum.adobe.com/react-spectrum/IllustratedMessage.html) from the spectrum.

**Review by KelvinOm — DISMISSED** [review]

File: app/client/src/pages/Editor/IDE/PagesPane/EmptyState.tsx (hunk 1 above)
Comment (albinAppsmith) [reply app/client/src/pages/Editor/IDE/PagesPane/EmptyState.tsx:11]: For now this is not generalized. It will be only used for sidebars.

**Review by hetunandu — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**